Fix passing custom APT_MIRROR on building Docker

cpuguy83 opened the pull request:
This reverts commit 4aa9b9c5b3da4c6d2478726824dbbfa2b1f5d1a7.

Commit breaks builds from the Makefile which do not consume build-args.
Specifically breaking `.ensure-syscall-test` for me because I have `DOCKER_BUILD_ARGS` set for a custom `APT_MIRROR`

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "revert_makefile_build_args" git@github.com:cpuguy83/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "revert_makefile_build_args" git@github.com:cpuguy83/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

hqhq commented:
@cpuguy83 How did you build from the Makefile, IOW how did you use `DOCKER_BUILD_ARGS` before and why commit `4aa9b9c` breaks it? Sorry I didn't get it, can you be more specific?

Thanks.

tonistiigi commented:
I think we should instead move `APT_MIRROR` out of `DOCKER_BUILD_ARGS` and invoke it with `make BUILD_APT_MIRROR=foo`.

cpuguy83 commented:
@hqhq Because build args are being passed down to all the integration test fixtures and not ever consumed by those Dockerfiles.

cpuguy83 commented:
@tonistiigi Wouldn't it make more sense to pass through HTTP_PROXY?

cpuguy83 committed:
Pass through build APT_MIRROR as separate var
DOCKER_BUILD_APT_MIRROR

This allows passthrough of other options like HTTP_PROXY from
DOCKER_BUILD_ARGS to work on all builds (for example,
ensure-syscall-test) when APT_MIRROR is set.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

cpuguy83 commented:
In any case, I've changed this to use `DOCKER_BUILD_APT_MIRROR` instead of reverting.

tonistiigi commented:
LGTM

vdemeester commented:
LGTM 🐸
